File: padroesCriacionais/padrao-de-projeto-builder/php/src/UsuarioBuilder.php
<?php

namespace Builder\Php;

class UsuarioBuilder
{
    private Usuario $usuario;

    public function __construct()
    {
        $this->usuario = new Usuario();
    }

    public function build(): Usuario
    {
        if($this->usuario->getNome() === '') {
            throw new \InvalidArgumentException('Nome inválido');
        }

        if($this->usuario->getEmail() === '') {
            throw new \InvalidArgumentException('Email inválido');
        }

        if($this->usuario->getSenha() === '') {
            throw new \InvalidArgumentException('Senha inválida');
        }

        return $this->usuario;
    }

    public function setNome(string $nome): UsuarioBuilder
    {
        $this->usuario->setNome($nome);
        return $this;
    }

    public function setEmail(string $email): UsuarioBuilder
    {
        $this->usuario->setEmail($email);
        return $this;
    }

    public function setSenha(string $senha): UsuarioBuilder
    {
        $this->usuario->setSenha($senha);
        return $this;
    }

    public function setEndereco(string $endereco): UsuarioBuilder
    {
        $this->usuario->setEndereco($endereco);
        return $this;
    }

    public function setApelido(string $apelido): UsuarioBuilder
    {
        $this->usuario->setApelido($apelido);
        return $this;
    }

    public function setDicaSenha(string $dicaSenha): UsuarioBuilder
    {
        $this->usuario->setDicaSenha($dicaSenha);
        return $this;
    }

}

File: padroesCriacionais/padrao-de-projeto-builder/java/builder/src/main/java/UsuarioBuilder.java
import org.example.Usuario;

public class UsuarioBuilder {

    private Usuario usuario = null;
    public UsuarioBuilder() {
        this.usuario = new Usuario();
    }

    public Usuario build()
    {
        if(this.usuario.getNome().equals("")) {
            throw new IllegalArgumentException("Nome inválido");
        }

        if(this.usuario.getEmail().equals("")) {
            throw new IllegalArgumentException("Email inválido");
        }

        if(this.usuario.getSenha().equals("")) {
            throw new IllegalArgumentException("Senha inválida");
        }

        return this.usuario;
    }

    public UsuarioBuilder setNome(String nome) {
        this.usuario.setNome(nome);
        return this;
    }

    public UsuarioBuilder setEmail(String email) {
        this.usuario.setEmail(email);
        return this;
    }

    public UsuarioBuilder setSenha(String senha) {
        this.usuario.setSenha(senha);
        return this;
    }

    public UsuarioBuilder setEndereco(String endereco) {
        this.usuario.setEndereco(endereco);
        return this;
    }

    public UsuarioBuilder setApelido(String apelido) {
        this.usuario.setApelido(apelido);
        return this;
    }

    public UsuarioBuilder setDicaSenha(String dicaSenha) {
        this.usuario.setDicaSenha(dicaSenha);
        return this;
    }
}
